pkg/csi/service/common/util.go

```diff
@@ -198,17 +200,54 @@ func ParseStorageClassParams(ctx context.Context, params map[string]string) (*St
 		DatastoreURL:      "",
 		StoragePolicyName: "",
 	}
+	if !CSIMigrationFeatureEnabled {
```

Comment:
Would be nice to have a unit test for this.

Reply:
Added migration related unit tests for `ParseStorageClassParams`

```go
func TestParseStorageClassParamsWithMigrationEnabled(t *testing.T) {
	CSIMigrationFeatureEnabled = true
	params := map[string]string{
		CSIMigrationParams:                   "true",
		DatastoreMigrationParam:              "vSANDatastore",
		AttributeStoragePolicyName:           "policy1",
		HostFailuresToTolerateMigrationParam: "1",
		ForceProvisioningMigrationParam:      "true",
		CacheReservationMigrationParam:       " 25",
		DiskstripesMigrationParam:            "2",
		ObjectspacereservationMigrationParam: "50",
		IopslimitMigrationParam:              "16",
	}
	expectedScParams := &StorageClassParams{
		CSIMigration:      "true",
		StoragePolicyName: "policy1",
		Datastore:         "vSANDatastore",
	}
	actualScParams, err := ParseStorageClassParams(ctx, params)
	if err != nil {
		t.Errorf("failed to parse params: %+v", params)
	}
	if !isStorageClassParamsEqual(expectedScParams, actualScParams) {
		t.Errorf("Expected: %+v\n Actual: %+v", expectedScParams, actualScParams)
	}
}

func TestParseStorageClassParamsWithMigrationDisabled(t *testing.T) {
	CSIMigrationFeatureEnabled = false
	params := map[string]string{
		CSIMigrationParams:                   "true",
		DatastoreMigrationParam:              "vSANDatastore",
		AttributeStoragePolicyName:           "policy1",
		HostFailuresToTolerateMigrationParam: "1",
	}
	actualScParams, err := ParseStorageClassParams(ctx, params)
	if err == nil {
		t.Errorf("error expected but not received. actualScParams: %v", actualScParams)
	}
	t.Logf("expected err received. err: %v", err)
}
```


both tests passed.

```
=== RUN   TestParseStorageClassParamsWithMigrationEnabled
{"level":"info","time":"2020-06-19T09:07:11.217726-07:00","caller":"common/util.go:228","msg":"vSphere CSI Driver supports StoragePolicyName, Datastore and fsType parameters supplied from legacy in-tree provisioner. All other parameters supplied in csimigrationparams will be dropped."}
--- PASS: TestParseStorageClassParamsWithMigrationEnabled (0.00s)
=== RUN   TestParseStorageClassParamsWithMigrationDisabled
--- PASS: TestParseStorageClassParamsWithMigrationDisabled (0.00s)
    util_test.go:345: expected err received. err: Invalid param: "csimigration" and value: "true"

```